iaf-commons/src/main/java/nl/nn/adapterframework/util/StringUtil.java

```diff
@@ -283,4 +287,25 @@ public static Stream<String> splitToStream(@Nullable final String input, @Nonnul
 		return splitPattern.splitAsStream(input.trim())
 				.filter(StringUtils::isNotBlank);
 	}
+
+	/**
+	 * toStrings and concatenates all fields of the given object, except fields containing the word 'password' or 'secret'.
+	 * 'fail-safe' method, returns toString if it is unable to use reflection.
+	 * Uses the {@link #OMIT_PASSWORD_FIELDS_STYLE OMIT_PASSWORD_FIELDS_STYLE}.
+	 * 
+	 * @see org.apache.commons.lang3.builder.ToStringBuilder#reflectionToString
+	 */
+	@Nonnull
+	public static String reflectionToString(@Nullable Object object) {
+		if(object == null) {
+			return "<null>";
+		}
+
+		try {
+			return new ReflectionToStringBuilder(object, OMIT_PASSWORD_FIELDS_STYLE).toString();
+		} catch (Exception e) { //amongst others, IllegalAccess-, ConcurrentModification- and Security-Exceptions
+			LogManager.getLogger(object).warn("exception getting string representation of object", e);
+			return object.toString();
```

Comment:
Dit lijkt mij een onveilige fallback!

Immers, de `toString()` van dit object is waarschijnlijk geimplementeerd door de reflectionToString aan te roepen! Dus dit geeft een recursieve aanroep, en dat nog wel in de exception-flow.

BRRRRRRRRRR!